build.gradle.kts

```diff
@@ -10,86 +9,115 @@ description = "A Result monad for modelling success or failure operations."
 plugins {
     `maven-publish`
     signing
-    kotlin("jvm") version "1.3.72"
+    kotlin("multiplatform") version "1.3.72"
     id("org.jetbrains.dokka") version "0.10.1"
     id("com.github.ben-manes.versions") version "0.28.0"
     id("net.researchgate.release") version "2.8.1"
     id("kotlinx.benchmark") version "0.2.0-dev-8"
     id("org.jetbrains.kotlin.plugin.allopen") version "1.3.72"
 }
 
+tasks.withType<DependencyUpdatesTask> {
+    rejectVersionIf {
+        listOf("alpha", "beta", "rc", "cr", "m", "eap", "pr").any {
+            candidate.version.contains(it, ignoreCase = true)
+        }
+    }
+}
+
+val dokka by tasks.existing(DokkaTask::class) {
+    outputFormat = "javadoc"
+    outputDirectory = "$buildDir/docs/javadoc"
+}
+
+val javadocJar by tasks.registering(Jar::class) {
+    group = LifecycleBasePlugin.BUILD_GROUP
+    description = "Assembles a jar archive containing the Javadoc API documentation."
+    archiveClassifier.set("javadoc")
+    dependsOn(dokka)
+    from(dokka.get().outputDirectory)
+}
+
 allprojects {
     repositories {
         mavenCentral()
         jcenter()
         maven("https://dl.bintray.com/kotlin/kotlinx")
     }
-
-    tasks.withType<KotlinCompile> {
-        kotlinOptions {
-            jvmTarget = "1.8"
-            freeCompilerArgs = listOf("-Xuse-experimental=kotlin.contracts.ExperimentalContracts")
-        }
-    }
 }
 
 allOpen {
     annotation("org.openjdk.jmh.annotations.State")
     annotation("org.openjdk.jmh.annotations.BenchmarkMode")
 }
 
-sourceSets.create("benchmark") {
-    java {
-        srcDir("src/benchmarks/kotlin")
-    }
-}
-
-val benchmarkImplementation by configurations
+// creating this sourceSet and specifying jvm { withJava() } will create a
+// "jvmBenchmark" sourceSet we can access from kotlin mpp extension
+sourceSets.create("benchmark")
 
 benchmark {
     targets {
-        register("benchmark")
+        register("jvmBenchmark")
     }
 }
 
-dependencies {
-    implementation(kotlin("stdlib"))
-    testImplementation("junit:junit:4.13")
-    testImplementation(kotlin("test-common"))
-    testImplementation(kotlin("test-annotations-common"))
-    testImplementation(kotlin("test-junit"))
-    testImplementation(kotlin("test"))
+kotlin {
+    jvm {
+        withJava()
+    }
+    sourceSets {
+        val commonMain by getting {
```

Comment:
Can we make this use the `named` methods, just like I do here: https://github.com/michaelbull/kotlin-inline-logger/blob/master/build.gradle.kts#L46

Otherwise IntelliJ moans that these `val`s (commonMain, commonTest, etc) are unused variables.

Reply:
ill take a look, it made specifying that `jvmBenchmark dependsOn(jvmMain)` nice and readable

Reply:
Oh if you are actually using the `jvmMain` as a variable then ignore me!